"Diff" GitHub action should handle ServiceDirectory changes

#5686 failed in the Diff GitHub Action with this log:

```text
Detecting changes for ServiceDirectory
ServiceDirectory is new
Checking compatibility with previous releases
Configuration error: No API with ID 'ServiceDirectory'
Error: Process completed with exit code 1.
```

We should sort that out...

--- a/detect-pr-changes.sh
+++ b/detect-pr-changes.sh
@@ -7,7 +7,8 @@ log_header() {
   echo -e "\e[1;35m$1\e[0m"
 }
 
-apis=$(git diff master --name-only | grep -e 'apis/.*/' | cut -d/ -f 2 | uniq)
+# Find the APIs that have changed, excluding ServiceDirectory (which isn't a real API)
+apis=$(git diff master --name-only | grep -e 'apis/.*/' | cut -d/ -f 2 | grep -v '^ServiceDirectory$' | uniq)
 
 if [[ "$apis" == "" ]]
 then

